src/screen.js

```diff
@@ -2,51 +2,51 @@
 
 /**
  *  Screen handles calculator screen.  
- *  Constructor.
- *  @param {String} element.
  */
-function Screen(element) {
-    this.element = element;
-    this.reset = false;
-}
-
-/**
- * Allows a new entry. 
- */
-Screen.prototype.resetOnNextInput = function() {
-    this.reset = true;
-}
-
-Screen.prototype.resetOfNextInput = function() {
-    this.reset = false;
-}
+class Screen {
+
+    /**
+     *  Constructor.  
+     *  @param {String} element.
+     */
+    constructor(element) {
+        this.element = element;
+        this.reset = false;
+    }
 
-/**
- * Sets the given number on the screen 
- * @param {Number} number
- */
-Screen.prototype.setNumber = function(number) {
-    this.element.value = number;
-}
+    /**
+     * Allows a new entry. 
+     */
+    resetOnNextInput() {
+        this.reset = true;
+    }
 
-/**
- * Gets the current number from the screen. 
- */
-Screen.prototype.getNumber = function() {
-    return this.element.value;
-};
+    /**
+     * Sets the given number on the screen 
+     * @param {Number} number
+     */
+    setNumber(number) {
+        this.element.value = number;
+    }
 
-/**
- * Adds new number on the screen. 
- * @param {Number} digit.
- */
-Screen.prototype.addDigit = function(digit) {
+    /**
+     * Gets the current number from the screen. 
+     */
+    getNumber() {
+        return this.element.value;
```

Comment:
As I wrote above, here is the problem. If you have a method `getNumber()` it needs to return a number, not string, since that becomes a poison for the calling code:
```
return Number(this.element.value);
```